src/ot_orchestration/operators/manifest_operators.py

```diff
@@ -0,0 +1,284 @@
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.staging_manifest_path_pattern = staging_manifest_path_pattern
+        self.gcp_conn_id = gcp_conn_id
+
+    def execute(self, **_) -> list[Manifest_Object]:
+        """Read manifests.
+
+        Raises:
+            ValueError: when incorrect glob is defined.
+            NotImplementedError: for protocol other then gs.
+
+        Returns:
+            list[Manifest_Object]: list of read manifests.
+        """
+        logging.info(self.staging_manifest_path_pattern)
+        pattern = r"^((?P<protocol>.*)://)?(?P<root>[(\w)-]+)/(?P<prefix>([(\w)-/])+?)/(?P<matchglob>[(\w)-*]+.*){1}"
+        compiled_pattern = re.compile(pattern)
+        _match = compiled_pattern.match(self.staging_manifest_path_pattern)
+        if _match is None:
+            raise ValueError(
+                "Incorrect glob pattern %s", self.staging_manifest_path_pattern
+            )
+        protocol = _match.group("protocol")
+        root = _match.group("root")
+        prefix = _match.group("prefix")
+        matchglob = _match.group("matchglob")
+
+        if protocol != "gs":
+            raise NotImplementedError(
+                "Listing objects from path with %s protocol is not implemented",
+                protocol,
+            )
+        logging.info(
+            "Listing files at %s/%s with match glob %s", root, prefix, matchglob
+        )
+        manifest_paths = GCSHook(gcp_conn_id=self.gcp_conn_id).list(
+            bucket_name=root,
+            prefix=prefix,
+            match_glob=matchglob,
+        )
+        manifest_paths = [f"{protocol}://{root}/{mp}" for mp in manifest_paths]
+        manifests = IOManager().load_many(manifest_paths)
+        return manifests
+
+
+class ManifestSubmitBatchJobOperator(BaseOperator):
+    """Submit manifest as a batch job."""
+
+    template_fields: Sequence[str] = ["job_name", "manifests", "step"]
+
+    def __init__(
+        self, step: str, job_name: str, manifests: list[Manifest_Object], **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.job_name = job_name
+        self.kwargs = kwargs
+        self.manifests = manifests
+        self.step = step
+
+    def execute(self, context: Context) -> str:
+        """Execute the operator.
+
+        Args:
+            context (Context): execution context
+
+        Raises:
+            AirflowSkipException: when no manifest files are found
+
+        Returns:
+            str: google batch job name.
+        """
+        # in case no manifest files, do not create the batch job
+        manifest_paths = [m["manifestPath"] for m in self.manifests]
+
+        if not manifest_paths:
+            raise AirflowSkipException("No manifests to run the batch job")
+        params = context.get("params")
+        gcp = params.get("gcp")
+        gcp_project = gcp.get("GCP_PROJECT")
+        gcp_region = gcp.get("GCP_REGION")
```

Comment:
In case of using config files it is fairly easy to plug them in. I imagine that this will not change so much for our platform, it may be required if someone wants to reuse it ... in far future. Lets keep one place in the library for now (will decrease the complexity). We can always extract it to the config if needed by the community.